### PR TITLE
ci: replace travis with github action

### PR DESCRIPTION
**Pull request by inigomarquinez**
This PR migrates the CI pipeline from travis to GitHub actions and adds node version 11 to 22 to the matrix.

The travis project should be removed after merging the PR (see [here](https://github.com/inigomarquinez/expressjs-multer/actions/runs/9034165622)).

The CI pipeline works in my local fork.

Related to https://github.com/jshttp/http-errors/issues/108

### CODE REVIEW

**Review by UlisesGascon — APPROVED** [review]
LGTM!

File: package.json [hunk 1]
```diff
@@ -48,6 +49,9 @@
     "lib/"
   ],
   "scripts": {
-    "test": "standard && mocha"
+    "lint": "standard",
+    "test": "mocha --reporter spec --bail --exit --check-leaks test/",
```
Comment (ctcpip) [inline package.json:53]:
```suggestion
    "test": "standard && mocha --reporter spec --bail --exit --check-leaks test/",
```

**Review by ctcpip — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -33,7 +33,8 @@
     "express": "^4.13.1",
     "form-data": "^1.0.0-rc1",
     "fs-temp": "^1.1.2",
-    "mocha": "^3.5.3",
+    "mocha": "^10.4.0",
+    "nyc": "^15.1.0",
```
Comment (ctcpip) [inline package.json:37]: these should be devDependencies, along with standard and maybe others

**Review by ctcpip — CHANGES_REQUESTED** [review]
`package-lock.json`  should not be added

**Comment by inigomarquinez** [comment]
@UlisesGascon , @ctcpip 

Is this PR ready to be merged? Do you need me to do something else?

**Review by ctcpip — APPROVED** [review]

**Pull request merged**